.github/workflows/check_python_dependencies.yml

```
@@ -0,0 +1,47 @@
+name: Check Python Dependencies
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  check_dependencies:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+                    
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      
+      - name: Install repo-tools
+        run: pip install git+https://github.com/salman2013/repo-tools.git@salman/add-script-for-python-dependencies
+      
+      - name: Install requirements parser module
+        run: pip install requirements-parser  
+      
+      - name: Install setuptool
+        run: pip install setuptools  
```

```suggestion
      - name: Install repo-tools
        run: pip install edx-repo-tools[find_dependencies]
```

I think this is all you should need now.

---

@feanil repo-tool is requiring `setuptool` to run this github action on `3.12` so we have two options either we should keep it here like following step or install it in `repo-tool` but I believe it should be here where we need it as python has dropped its default support in 3.12

> - name: Install setuptool
        run: pip install setuptools

---

I have updated other things and everything is working fine.